### PR TITLE
Add option to disable cargo-workspaces commit message branding

### PR DESCRIPTION
**Pull request by foresterre**
Hello 👋,

I have been using cargo-workspaces for a while now and it has been a great tool so far. There is one thing I don't really like, which is the line `Generated by cargo-workspaces` added to each commit produced by the tool (I think it 'pollutes', for lack of a better word, the git history log). 

This change would add the option to not include the 'Generated by cargo-workspaces' message.

### CODE REVIEW

File: cargo-workspaces/src/utils/git.rs [hunk 1]
```diff
@@ -45,6 +45,10 @@ pub struct GitOpt {
     #[clap(short, long, conflicts_with_all = &["amend"])]
     pub message: Option<String>,
 
+    /// Do not add 'Generated by cargo-workspaces' to the commit message
```
Comment (pksunkara) [inline cargo-workspaces/src/utils/git.rs:48]:
```suggestion
    /// Do not add branding to the commit message
```

File: cargo-workspaces/src/utils/git.rs [hunk 2]
```diff
@@ -45,6 +45,10 @@ pub struct GitOpt {
...
+    #[clap(long)]
+    pub no_git_commit_msg_branding: bool,
```
Comment (pksunkara) [inline cargo-workspaces/src/utils/git.rs:50]:
```suggestion
    pub no_branding: bool,
```

File: cargo-workspaces/src/utils/git.rs [hunk 3]
```diff
@@ -235,14 +239,21 @@ impl GitOpt {
...
-            "{}\n\n{}\n\nGenerated by cargo-workspaces",
+            "{}\n\n{}{}",
             msg,
             new_versions
                 .iter()
                 .map(|x| format!("{}@{}", x.0, x.1))
                 .collect::<Vec<_>>()
-                .join("\n")
+                .join("\n"),
+            postfix
```
Comment (pksunkara) [inline cargo-workspaces/src/utils/git.rs:256]: Let's rename this variable to `branding`

**Review by pksunkara — CHANGES_REQUESTED** [review]
Initially, I wanted to merge this PR after the review comments are addressed. But then I started thinking. There had been several times where I found that having something clearly marked as generated is useful and not being marked as generated leading to some unintended consequences. 

---

> I think it 'pollutes', for lack of a better word, the git history log

This is very deceitful. How does it pollute the git history? It doesn't create an extra commit with the branding. When you do `git log` it doesn't show up as an extra commit. It shows up a commit with the proper description with versions.

Even if you were to remove the branding, you would still have the version commit. So, according to you, we still haven't solved the pollution problem.

File: cargo-workspaces/src/utils/git.rs [hunk 4]
```diff
@@ -29,7 +29,7 @@ pub struct GitOpt {
     #[clap(long, conflicts_with_all = &[
         "allow-branch", "amend", "message", "no-git-tag",
         "tag-prefix", "individual-tag-prefix", "no-individual-tags",
-        "no-git-push", "git-remote"
+        "no-git-push", "git-remote", "no-git-commit-msg-branding"
```
Comment (pksunkara) [inline cargo-workspaces/src/utils/git.rs:32]: ?

Comment (foresterre) [reply cargo-workspaces/src/utils/git.rs]: Haha, yes, you were too quick for me; I saw this too :)

**Comment by foresterre** [comment]
Perhaps calling it 'pollutes' was a bit too harsh, but I couldn't find the right word. My apologies for that :).

I think having the commit clearly marked could be favourable in some cases.
This PR will not enable it by default, but adds an explicit option, so if you do like commits to be marked as generated by cargo-workspaces, you (as a user of this tool) don't have to do anything :). If you do prefer to have the branding removed, you can use the extra option. 

Of course, if you prefer to keep the branding for all users, feel free to close this PR :smile:.

**Comment by pksunkara** [comment]
I had a really long thinking session on this yesterday. I think I want to have the branding. And if anyone really doesn't want it, they could always fork the repo and modify the code and use the modified tool. Nothing stopping them. :)

I really appreciate you making a PR though. Hopefully I was responsive enough.

**Pull request closed without merging**